src/Aspire.Hosting.Azure.ServiceBus/AzureServiceBusExtensions.cs

```diff
@@ -122,21 +132,373 @@ public static IResourceBuilder<AzureServiceBusResource> AddQueue(this IResourceB
+                    {
+                        throw new InvalidOperationException("The configuration file mount could not be parsed.");
+                    }
+                    annotation.Configure(jsonObject);
+                    jsonObject.WriteTo(writer);
+                }
+            }
+        });
+
+        var healthCheckKey = $"{builder.Resource.Name}_check";
+
+        if (configureContainer != null)
+        {
+            var surrogate = new AzureServiceBusEmulatorResource(builder.Resource);
+            var surrogateBuilder = builder.ApplicationBuilder.CreateResourceBuilder(surrogate);
+            configureContainer(surrogateBuilder);
+        }
+
+        // To use the existing ServiceBus health check we would need to know if there is any queue or topic defined.
+        // We can register a health check for a queue and then no-op if there are no queues. Same for topics.
+        // If no queues or no topics are defined then the health check will be successful.
+
+        builder.ApplicationBuilder.Services.AddHealthChecks()
+          .Add(new HealthCheckRegistration(
+              healthCheckKey,
+              sp => new ServiceBusHealthCheck(
+                  () => serviceBusClient ?? throw new DistributedApplicationException($"{nameof(serviceBusClient)} was not initialized."),
+                  () => queueOrTopicName),
+              failureStatus: default,
+              tags: default,
+              timeout: default));
+
+        builder.WithHealthCheck(healthCheckKey);
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds a bind mount for the data folder to an Azure Service Bus emulator resource.
+    /// </summary>
+    /// <param name="builder">The builder for the <see cref="AzureServiceBusEmulatorResource"/>.</param>
+    /// <param name="path">Relative path to the AppHost where emulator storage is persisted between runs. Defaults to the path '.servicebus/{builder.Resource.Name}'</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureServiceBusEmulatorResource> WithDataBindMount(this IResourceBuilder<AzureServiceBusEmulatorResource> builder, string? path = null)
+        => builder.WithBindMount(path ?? $".servicebus/{builder.Resource.Name}", "/data", isReadOnly: false);
+
+    /// <summary>
+    /// Adds a named volume for the data folder to an Azure Service Bus emulator resource.
+    /// </summary>
+    /// <param name="builder">The builder for the <see cref="AzureServiceBusEmulatorResource"/>.</param>
+    /// <param name="name">The name of the volume. Defaults to an auto-generated name based on the application and resource names.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureServiceBusEmulatorResource> WithDataVolume(this IResourceBuilder<AzureServiceBusEmulatorResource> builder, string? name = null)
```

Comment:
Is `WithDataVolume` still valid? I didn't think the data got perserved across runs.

Reply:
But I removed it ... my brain thinks I did, what did I remove then